extensions/amp-date-picker/1.0/component/base-date-picker.tsx

```diff
@@ -0,0 +1,42 @@
+import {
+  DayPicker,
+  DayPickerProps,
+  SelectRangeEventHandler,
+  SelectSingleEventHandler,
+} from 'react-day-picker';
+
+import * as Preact from '#preact';
+
+import {DayButton} from './day-button';
+import {useDatePickerContext} from './use-date-picker-context';
+
+interface BaseDatePickerProps extends DayPickerProps {
+  onSelect: SelectSingleEventHandler | SelectRangeEventHandler;
```

Comment:
`onSelect` isn't included in `DayPickerProps`. React day picker has type definitions for single date picker and range date picker that extend `DayPickerProps` and add their own type for `onSelect`, but since this base is for both types I didn't want to use one or the other of those types so I just added the `onSelect` attribute.

Reply:
I ended up changing this type as well just to be a union type of `DayPickerSingleProps` and `DayPickerRangeProps`. Typescript is smart enough to know which type to match based on the `type` prop.